OrionUO/CMakeLists.txt

```diff
@@ -415,8 +414,8 @@ else()
     set_target_properties(orion PROPERTIES COTIRE_CXX_PREFIX_HEADER_INIT "stdafx.h")
     cotire(orion)
     if(ORION_OSX)
-      add_custom_command(TARGET orion POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy "liborion.dylib" "${CMAKE_BINARY_DIR}/orion.so")
-      add_custom_command(TARGET orion_unity POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy "unity/liborion.dylib" "${CMAKE_BINARY_DIR}/orion.so")
+      add_custom_command(TARGET orion POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_CFG_INTDIR}/liborion.dylib" "${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_CFG_INTDIR}/orion.so")
+      add_custom_command(TARGET orion_unity POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_CFG_INTDIR}/unity/liborion.dylib" "${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_CFG_INTDIR}/orion.so")
     else()
       add_custom_command(TARGET orion POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy "liborion.so" "${CMAKE_BINARY_DIR}/orion.so")
       add_custom_command(TARGET orion_unity POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy "unity/liborion.so" "${CMAKE_BINARY_DIR}/orion.so")
```

Comment:
Make sense. On the other hand, i’m not sure how to wrap it, since whole main loop need redesign. This was just first step to make it at least playable on mac/linux. Maybe keep whole set of changes under some define or new branch? 

There is no advantage of using “timers” and sdl events in main loop. We can keep it simple with delays per action.

Ad clang - ok ;)

Reply:
Ifdefing the whole main loop should be ok for now, no problem there.

I would also like to understand what is happening to your build, as on my end both windows and linux build are pretty much playable with constant 32fps with the timerthread (even running in debug). What is your CPU? 

I've tested your changes yesterday and didn't see any other difference than this implementation makes the fps unstable (I got variations from 26-33fps), as this implementation does not have any kind of frame control (frameskip or lock).

But either way, if this fixes your issue I do not see much trouble having a bit of fps instability right now, but at least I would want to find the reason you have this stutter.

Reply:
Previous loop was working only if there is not much input from server side. Once there is a lot of dynamics object it start to be unplayable. You can see it here
https://youtu.be/A0fs9mi5hMI. With this simple change i can run everything without any problems... at least for some time. After 30+min of real game usage it starts to be unplayable also, but for totally different reason (i'll debug why).

Reply:
Ok, so I retested in a crowded server and still doesn't have the issue here. I see in your video your fps is not locked, so it may explain a bit, and as I said mine is sync at 30fps and no issues at all. Then I retried your version in the same server and got a lot of stuttering and lower fps :(
So while we can't be sure this works for everyone, lets use the defines until we can fix this for real.

Reply:
Alright, I tested with fps unlocked and I can reproduce it. and yes, this is caused by the network activity with both versions but yours is a better with higher fps, but worse with lower fps.
So we definitively want the defines, but let make your version ON.

Do this and we can merge, thanks!

Reply:
:) This is just first version with removed timer. It does need some serious redesign which i'll do. There is so many redundant call. All try to remove them all.

Reply:
Try it now :) It should work much better. With this basic loop i'll have a loot at profiler and check what can we optimise to lower cpu usage.